api/http/api_inventory.go

```diff
@@ -384,37 +388,85 @@ func (i *inventoryHandlers) AddDeviceHandler(w rest.ResponseWriter, r *rest.Requ
 
 func (i *inventoryHandlers) UpdateDeviceAttributesHandler(w rest.ResponseWriter, r *rest.Request) {
 	ctx := r.Context()
-
 	l := log.FromContext(ctx)
-
 	//get device ID from JWT token
 	idata, err := identity.ExtractIdentityFromHeaders(r.Header)
 	if err != nil {
 		u.RestErrWithLogMsg(w, r, l, err, http.StatusUnauthorized, "unauthorized")
 		return
 	}
-
+	deviceID := model.DeviceID(idata.Subject)
 	//extract attributes from body
 	attrs, err := parseAttributes(r)
 	if err != nil {
 		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
 		return
 	}
+	i.updateDeviceAttributes(w, r, ctx, attrs, deviceID, model.AttrScopeInventory, nil)
+}
+
+func (i *inventoryHandlers) UpdateDeviceTagsHandler(w rest.ResponseWriter, r *rest.Request) {
+	ctx := r.Context()
+	l := log.FromContext(ctx)
+
+	// get device ID from uri depending from a scope
+	deviceID := model.DeviceID(r.PathParam("id"))
+	if len(deviceID) < 1 {
+		u.RestErrWithLog(w, r, l, errors.New("device id cannot be empty"), http.StatusBadRequest)
+		return
+	}
+
+	ifMatchHeader := r.Header.Get("If-Match")
+
+	// extract attributes from body
+	attrs, err := parseAttributes(r)
+	if err != nil {
+		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		return
+	}
+
+	// set scope and timestamp for tags attributes
+	now := time.Now()
+	for i := range attrs {
+		attrs[i].Scope = model.AttrScopeTags
+		if attrs[i].Timestamp == nil {
+			attrs[i].Timestamp = &now
+		}
+	}
+
+	i.updateDeviceAttributes(w, r, ctx, attrs, deviceID, model.AttrScopeTags, &ifMatchHeader)
+}
+
+func (i *inventoryHandlers) updateDeviceAttributes(
+	w rest.ResponseWriter,
+	r *rest.Request,
+	ctx context.Context,
+	attrs model.DeviceAttributes,
+	deviceID model.DeviceID,
+	scope string,
+	etag *string,
```

Comment:
Why is etag a pointer? I believe a string is enough, and you can check if not empty instead of `!= nil`

Reply:
Makes sense. Looks like left over from the beginning and not doesn't seems like it's needed. Thanks.